packages/mermaid/src/diagrams/xychart/xychartDb.ts

```diff
@@ -0,0 +1,302 @@
+          titlePadding: 5,
+          showTick: true,
+          tickLength: 5,
+          tickWidth: 2,
+        },
+        chartOrientation: 'vertical',
+        plotReservedSpacePercent: 50,
+      };
+}
+
+function getChartDefalutData(): XYChartData {
+  return {
+    yAxis: {
+      type: 'linear',
+      title: '',
+      min: Infinity,
+      max: -Infinity,
+    },
+    xAxis: {
+      type: 'band',
+      title: '',
+      categories: [],
+    },
+    title: '',
+    plots: [],
+  };
+}
+
+let xyChartConfig: XYChartConfig = getChartDefaultConfig();
+let xyChartThemeConfig: XYChartThemeConfig = getChartDefaultThemeConfig();
+let xyChartData: XYChartData = getChartDefalutData();
+let plotColorPalette = Array.isArray(xyChartThemeConfig.xychartPlotBaseColor)
+  ? xyChartThemeConfig.xychartPlotBaseColor
+  : plotColorPaletteGenerator(xyChartThemeConfig.xychartPlotBaseColor);
+let hasSetXAxis = false;
+let hasSetYAxis = false;
+
+interface NormalTextType {
+  type: 'text';
+  text: string;
+}
+
+function textSanitizer(text: string) {
+  return sanitizeText(text.trim(), config);
+}
+
+function parseDirective(statement: string, context: string, type: string) {
+  // @ts-ignore: TODO Fix ts errors
+  mermaidAPI.parseDirective(this, statement, context, type);
+}
+
+function setOrientation(oriantation: string) {
+  if (oriantation === 'horizontal') {
+    xyChartConfig.chartOrientation = 'horizontal';
+  } else {
+    xyChartConfig.chartOrientation = 'vertical';
+  }
+}
+function setXAxisTitle(title: NormalTextType) {
+  xyChartData.xAxis.title = textSanitizer(title.text);
+}
+function setXAxisRangeData(min: number, max: number) {
+  xyChartData.xAxis = { type: 'linear', title: xyChartData.xAxis.title, min, max };
+  hasSetXAxis = true;
+}
+function setXAxisBand(categories: NormalTextType[]) {
+  xyChartData.xAxis = {
+    type: 'band',
+    title: xyChartData.xAxis.title,
+    categories: categories.map((c) => textSanitizer(c.text)),
+  };
+  hasSetXAxis = true;
+}
+function setYAxisTitle(title: NormalTextType) {
+  xyChartData.yAxis.title = textSanitizer(title.text);
+}
+function setYAxisRangeData(min: number, max: number) {
+  xyChartData.yAxis = { type: 'linear', title: xyChartData.yAxis.title, min, max };
+  hasSetYAxis = true;
+}
+
+// this function does not set `hasSetYAxis` as there can be multiple data so we should calculate the range accordingly
+function setYAxisRangeFromPlotData(data: number[]) {
+  const minValue = Math.min(...data);
+  const maxValue = Math.max(...data);
+  const prevMinValue = isLinearAxisData(xyChartData.yAxis) ? xyChartData.yAxis.min : Infinity;
+  const prevMaxValue = isLinearAxisData(xyChartData.yAxis) ? xyChartData.yAxis.max : -Infinity;
+  xyChartData.yAxis = {
+    type: 'linear',
+    title: xyChartData.yAxis.title,
+    min: Math.min(prevMinValue, minValue),
+    max: Math.max(prevMaxValue, maxValue),
+  };
+}
+
+function transformDataWithOutCategory(data: number[]): SimplePlotDataType {
```

Comment:
```suggestion
function transformDataWithoutCategory(data: number[]): SimplePlotDataType {
```

Reply:
Thanks! for checking the PR so minutely, fixed.